app/src/main/java/co/touchlab/droidconandroid/shared/interactors/RsvpInteractor.java

```diff
@@ -18,44 +19,62 @@
 public class RsvpInteractor
 {
     private final DatabaseHelper helper;
+    private final RsvpRequest    request;
+    private final AppPrefs       appPrefs;
 
     @Inject
-    public RsvpInteractor(DatabaseHelper helper)
+    public RsvpInteractor(DatabaseHelper helper, RsvpRequest request, AppPrefs appPrefs)
     {
         this.helper = helper;
+        this.request = request;
+        this.appPrefs = appPrefs;
     }
 
     public Single<Event> addRsvp(Long eventId)
     {
         return helper.getEventForId(eventId)
                 .filter(event -> event != null && event.rsvpUuid == null)
-                .map(event -> setRsvp(event, UUID.randomUUID().toString()))
-                .flatMapSingle(this :: updateEvent);
+                .map(event -> setRsvp(event, appPrefs.getUserUniqueUuid()))
+                .flatMapSingle(this :: updateDatabase)
+                .flatMap(this :: updateBackend);
     }
 
     public Single<Event> removeRsvp(Long eventId)
     {
         return helper.getEventForId(eventId)
                 .filter(event -> event != null)
-                .map(event -> setRsvp(event, null)).flatMapSingle(this :: updateEvent);
+                .map(event -> setRsvp(event, null))
+                .flatMapSingle(this :: updateDatabase)
+                .flatMap(this :: updateBackend);
+    }
+
+    private Single<Event> updateDatabase(Event event)
+    {
+        return helper.updateEvent(event)
+                .andThen(Single.just(event));
     }
 
-    private Single<Event> updateEvent(Event event)
+    private Single<Event> updateBackend(Event event)
     {
-        return Single.fromCallable(() ->
+        if(event.rsvpUuid == null)
+        {
+            // implement persisted task here when ready
+            return request.unRsvp(event.id, appPrefs.getUserUniqueUuid())
+                    .flatMap(ignore -> {
```

Comment:
This flatmap is essentially just a doOnNext.

It might also make sense to add a try/catch to the analytics calls so that we dont crash due to analytics when things are otherwise working